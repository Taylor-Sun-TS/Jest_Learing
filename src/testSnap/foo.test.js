jest.mock("./foo");
const foo = require("./foo");

// foo is a mock function
foo.mockImplementation(() => 42);
foo();
// > 42

describe("test actual function", () => {
    it("test 1 + 2", () => {
        const foo = jest.requireActual('./foo');

        expect(foo(1, 2)).toBe(3);
    });
});

describe("test mock function", () => {

    test("test mock function implementation", () => {
        expect(foo()).toBe(42);
    });


    test("test snapshot 1", () => {
        const snapshot = `
        <a className="normal" href="http://www.test1.com" >
          test1
        </a>
        `;

        expect(snapshot).toMatchSnapshot();
    });


    test("test snapshot 2", () => {
        const snapshot = `<a className="normal" href="http://www.test2.com"> test2 </a>`;

        expect(snapshot).toMatchSnapshot();
    });

});
