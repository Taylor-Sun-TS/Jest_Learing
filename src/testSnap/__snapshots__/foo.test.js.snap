// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test mock function test snapshot 1 1`] = `
"
        <a className=\\"normal\\" href=\\"http://www.test1.com\\" >
          test1
        </a>
        "
`;

exports[`test mock function test snapshot 2 1`] = `"<a className=\\"normal\\" href=\\"http://www.test2.com\\"> test2 </a>"`;
